src/station_cmd.cpp

```diff
@@ -3144,15 +3144,23 @@ static void GetTileDesc_Station(TileIndex tile, TileDesc *td)
 		Owner tram_owner = INVALID_OWNER;
 		if (road_rt != INVALID_ROADTYPE) {
 			const RoadTypeInfo *rti = GetRoadTypeInfo(road_rt);
-			td->roadtype = rti->strings.name;
-			td->road_speed = rti->max_speed / 2;
+			td->road_desc.type = rti->strings.name;
+			td->road_desc.speed = rti->max_speed / 2;
+			if (rti->grffile[0] != nullptr) {
+				const GRFConfig* gc = GetGRFConfig(rti->grffile[0]->grfid);
+				td->road_desc.grf = gc->GetName();
+			}
 			road_owner = GetRoadOwner(tile, RTT_ROAD);
 		}
 
 		if (tram_rt != INVALID_ROADTYPE) {
 			const RoadTypeInfo *rti = GetRoadTypeInfo(tram_rt);
-			td->tramtype = rti->strings.name;
-			td->tram_speed = rti->max_speed / 2;
+			td->tram_desc.type = rti->strings.name;
+			td->tram_desc.speed = rti->max_speed / 2;
+			if (rti->grffile[0] != nullptr) {
+				const GRFConfig* gc = GetGRFConfig(rti->grffile[0]->grfid);
+				td->tram_desc.grf = gc->GetName();
+			}
```

Comment:
Getting very duplicatey. Perhaps a constructor might be worth it?

Reply:
Tried my best to simplify by implementing a helper function, let me know what you think!